In [21]:
import pandas as pd

In [24]:
## Total number of playgrounds
data = pd.read_csv("CityPlaygrounds.csv")

total_playgrounds = data.shape[0]

print(f"Total number of playgrounds: {total_playgrounds}")

Total number of playgrounds: 125


In [22]:
## Display the neighborhoods, highest to lowest, with their respective amount of playgrounds
data = pd.read_csv("CityPlaygrounds.csv", index_col="neighborhood")
counts = dict()
for index in data.index:
    if index not in counts:
        counts[index] = 1
    else:
        counts[index] += 2

sorted_counts = dict(sorted(counts.items(), key=lambda item: item[1], reverse=True))
print(sorted_counts)


{'Squirrel Hill South': 15, 'Beechview': 9, 'South Side Slopes': 9, 'Allegheny Center': 7, 'Highland Park': 7, 'Sheraden': 7, 'Beltzhoover': 7, 'Crawford-Roberts': 5, 'Hazelwood': 5, 'East Liberty': 5, 'Upper Lawrenceville': 5, 'South Oakland': 5, 'Elliott': 5, 'Carrick': 5, 'Mount Washington': 5, 'Central Northside': 3, 'South Side Flats': 3, 'Brighton Heights': 3, 'Brookline': 3, 'Lincoln-Lemington-Belmar': 3, 'Troy Hill': 3, 'Crafton Heights': 3, 'Garfield': 3, 'Greenfield': 3, 'Morningside': 3, 'Lincoln Place': 3, 'Manchester': 3, 'Marshall-Shadeland': 3, 'Bloomfield': 3, 'Bedford Dwellings': 1, 'Lower Lawrenceville': 1, 'Banksville': 1, 'Homewood North': 1, 'Bon Air': 1, 'Central Oakland': 1, 'Spring Garden': 1, 'Windgap': 1, 'Westwood': 1, 'Perry South': 1, 'Homewood West': 1, 'Strip District': 1, 'Stanton Heights': 1, 'West Oakland': 1, 'East Carnegie': 1, 'Duquesne Heights': 1, 'Esplen': 1, 'Fairywood': 1, 'Fineview': 1, 'Regent Square': 1, 'Allentown': 1, 'Hays': 1, 'Terrace V

In [23]:
## Display the number of neighborhoods with 1 or less playgrounds
data = pd.read_csv("CityPlaygrounds.csv")

playground_counts = data["neighborhood"].value_counts()

low_playground_neighborhoods = playground_counts[playground_counts <= 1]

num_low_playground_neighborhoods = len(low_playground_neighborhoods)

print("The number of neighbrhoods with 1 or less playgrounds:" + str(num_low_playground_neighborhoods))

The number of neighbrhoods with 1 or less playgrounds:39


**Conclusion**
-
Part of finding the "best" neighborhood in the city of Pittsburgh is finding the number of playgrounds each neighborhood has. I am basing my conclusion off of the one with the highest number of parks, and that is *Squirrel Hill South*.
39 playgrounds of the 125 have only 1 or less playground. Of all 125, only Squirrel Hill South has 15 with Beechview and South Side Slopes tying for second with 9. All in all, since Squirrel Hill South has the most playgrounds, it wins the title of "best" in this category.